### PR TITLE
Add parameter server_rack_arguments

### PR DESCRIPTION
**Pull request by treydock**
* Default is an empty array
* Elements are added to ARGV in Puppet server's rack config.ru

This is an ugly work around to segfaults that occur when running Puppet under Passenger in Ruby 1.8.7 (CentOS 6.x).  The only way I've found to work around the segfaults is to enable profiling by appending the "--profile" flag to ARGV in config.ru.

### CODE REVIEW

**Comment by ekohl** [comment]
:+1: I think the test failure is unrelated our tests are broken on puppet 2.7.

File: templates/server/config.ru.erb [hunk 1]
```diff
@@ -18,6 +18,9 @@ ARGV << "--rack"
 # to ~puppet/.puppet
 ARGV << "--confdir" << "<%= scope.lookupvar('::puppet::server_dir') %>"
 ARGV << "--vardir"  << "<%= scope.lookupvar('::puppet::server_vardir') %>"
+<% (scope.lookupvar('::puppet::server_rack_arguments') || []).each do |server_rack_argument| -%>
```
Comment (domcleal) [inline templates/server/config.ru.erb]: I think this is causing the test failure, as server_rack_arguments can be `:undefined` (or `:undef`)

**Comment by treydock** [comment]
@domcleal the option server_rack_arguments shouldn't be undefined since the default in puppet::params is an empty array.  Since the test failed in puppet::passenger spec my guess is the tests are behaving such that it's like calling `class { 'puppet::passenger': }` directly rather than what I'd assume to be the only public class, `class { 'puppet': }`.

I think I only way to handle "undef" properly in all versions of Puppet is to reference the value using instance variable which is defined in the template's calling class.  https://docs.puppetlabs.com/guides/templating.html#testing-for-undefined-variables

Will update.

**Comment by treydock** [comment]
PR updated.  Ran into issue after removing my local Gemfile.lock and running `bundle install` that is related to rspec-puppet now being at version 2.0.0.  The use of `subject` no longer works for the methods in `spec/lib/module_spec_helper.rb`, but instead the subject argument must be passed using `catalogue`.  The unit tests wouldn't pass without that change so hopefully it's okay that I've included that change in this PR.

**Comment by treydock** [comment]
Disregard my comments about issues with rspec-puppet.  I see now that my branch was not up-to-date with master and @domcleal already addressed that issue.  Updating PR yet again and sorry for the noise.

**Comment by mmoll** [comment]
thanks @treydock :tada:

**Pull request merged**